src/components/Posts/PostMenu.tsx

```diff
@@ -0,0 +1,130 @@
+import {
+  IconButton,
+  ListItemIcon,
+  ListItemText,
+  Menu,
+  MenuItem,
+} from "@mui/material";
+import React, { useState } from "react";
+import { useAppDispatch, useAppSelector } from "../../state/hooks";
+import { MoreVert, Edit, Delete, Link } from "@mui/icons-material";
+import axios from "axios";
+import { deletePost } from "../../state/slices/postsSlice";
+import { useNavigate } from "react-router-dom";
+
+type PostMenuProps = {
+  authorId: number;
+  postId: number;
+  isExpanded?: boolean;
+};
+
+const styles = {
+  listItemIcon: {
+    "&.MuiListItemIcon-root": { color: "black.main" },
+  },
```

Comment:
A small thing, instead of applying this style to the `ListItemIcon`, should we add it directly to the icons? That way we don't have to worry about the selector and can just say:
```suggestion
  icon: {
    color: "black.main",
  },
```